### PR TITLE
Scraper fixes

### PR DESCRIPTION
**Pull request by dpshelio**
The scraper was failing when asked the time for an URL that contained text in separators, for example in urls like: `data/%Y/%m/swap%m%d_%H%M%S` where it couldn't distinguish between `%mswap` and `%m` as the same time type.

### CODE REVIEW

File: sunpy/net/dataretriever/sources/norh.py [hunk 1]
```diff
@@ -28,32 +30,12 @@ def _get_url_for_timerange(self, timerange, **kwargs):
...
-        """
-
-        # default urllib password anonymous@ is not accepted by the NoRH FTP
-        # server. include an accepted password in base url
-        baseurl = 'ftp://anonymous:mozilla@example.com@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/'
+        baseurl = 'ftp://solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/%Y/%m/{freq}%y%m%d'
 
         # We allow queries with no Wavelength but error here so that the query
         # does not get passed to VSO and spit out garbage.
+        # NOTE: should not error on the function?
```
Comment (Cadair) [inline sunpy/net/dataretriever/sources/norh.py]: what function?

File: sunpy/net/dataretriever/sources/eve.py [hunk 2]
```diff
@@ -39,35 +42,34 @@ def _get_url_for_timerange(self, timerange, **kwargs):
...
-        date : Python datetime object
-
-        Returns
-        -------
-        URL : string
-        """
-        base_url = 'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/'
-        return urljoin(base_url,
-                                date.strftime('%Y/%Y%m%d') + '_EVE_L0CS_DIODES_1m.txt')
+        baseurl = ('http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/'
```
Comment (Punyaslok) [inline sunpy/net/dataretriever/sources/eve.py]: Here, there's a baseurl definition, and also in the next function. Wouldn't it be good to have a baseurl() function and define each baseurl once ?

File: sunpy/net/dataretriever/sources/norh.py [hunk 3]
```diff
@@ -28,29 +30,8 @@ def _get_url_for_timerange(self, timerange, **kwargs):
...
-        Returns
-        -------
-        string
-            The URL for the corresponding date.
-        """
-
-        # default urllib password anonymous@ is not accepted by the NoRH FTP
-        # server. include an accepted password in base url
-        baseurl = 'ftp://anonymous:mozilla@example.com@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/'
+        baseurl = 'ftp://solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/%Y/%m/{freq}%y%m%d'
```
Comment (Punyaslok) [inline sunpy/net/dataretriever/sources/norh.py]: Same baseurl argument

File: sunpy/net/dataretriever/sources/eve.py (hunk 2 above)
Comment (dpshelio) [inline sunpy/net/dataretriever/sources/eve.py]: Good finding 👍

**Pull request merged**